### DISCUSSION TITLE
SCT013 an ESP32V3 mit Tasmota 12.3.1

### DISCUSSION THREAD

**Discussion by Hasont** [arendst/Tasmota#17756] [Q&A] [answered]
Hello, I would like to use an ESP32S3 for my mobile home. The ESP should take over the heating control and display the power consumption. The temperature display and heating control via rule parameters already work quite well. I also got a display for it. Unfortunately, I cannot read out the SCT013 correctly on the ADC via the ADC CT Power parameter. I have the parameter ADCParam set to AdcParam 7,0,275,0.23,0. But I have a background noise or whatever you call it from 30mA to 240mA. When connecting electrical devices, a good value is always added. So it seems to work in general. The SCT013 (10A/1V) is connectet via 2x10K and 10uF as described in the forums. What can I do and is it even possible to use the SCT013 with Tasmota. With an ESP8266 Amica and also on the ESP32 without a display, I unfortunately get the same result.
This text was translated from German into English.
Greetings from Franconia, Horst

**Comment by barbudor** [comment, marked as answer]
I don't have successfull experience with CT so I would always advise something else.

Is CT the only option ?
Why not using a PZEM ? or another ESP device like a Shelly or a Sonoff POW that have dedicated power measurment ?

**Reply by Hasont** [reply]
Hello Barbudor, if I switch to PZEM, could I also take the 10A version and then adapt it, or is the 100A version the only option?

**Reply by barbudor** [reply]
I think that you shouldn't attempt to modify it
If 10A is not enough you have to go for the 100A model
How much do you need ?

The Shelly EM can have up to 2 CT of either 50A or 120A but that's more expensive
Sonoff POW Elite or Origin can handle 16A or 20A for a little more money than the PZEM but they are in-line devices not CT.

**Reply by Hasont** [reply]
Thank you for your help. It is for a mobile home and since the PZEM is available in 10A and 100A the question is what Tasmota supports here. The 10A PZEM version would be enough for me in the camper.
Kind regards Horst

**Reply by barbudor** [reply]
Both works fine with tasmota

**Comment by Hasont** [comment]
Hello, thank you for your answer, I have little space in the mobile home and also no constant WiFi and electricity. So I would like to stick with the DSP32. The idea with the PZEM would still be a possibility if no other idea came up.
Kind regards Horst

**Reply by octavvian** [reply]
Hello,
maybe this will help ...
https://savjee.be/blog/Home-Energy-Monitor-ESP32-CT-Sensor-Emonlib/
Good luck!

**Reply by Hasont** [reply]
Thanks for the answer but that's exactly what I did. That's why I can't get any further. I have to try it without a mobile phone charger and without WLAN. The spikes (noise) have to come from somewhere.

**Reply by octavvian** [reply]
OK. I'm using that (SCT013 50A), but on a LANController (tinycontrol.eu) unit and works just fine.
Sorry, I'm off topic.

**Comment by DebrajKundu312** [comment]
Hey i want to use Tasmota with SCT013 sensor with esp32, How will i add firmware support for the sensor to show the current result on the tasmota web?
